tests/VCR/LibraryHooks/SoapHookTest.php

```diff
@@ -89,4 +92,19 @@ protected function getHeaderCheckCallback($expectedHeader)
             return new Response(200, array(), '');
         };
     }
+
+    /**
+     * @param array $expectedHeaders
+     * @return \callable
+     */
+    protected function getHeadersCheckCallback(array $expectedHeaders)
```

Comment:
Good idea. 
The other method `getHeaderCheckCallback` is now not necessary anymore. To avoid duplication of this logic, could you please use your new `getHeadersCheckCallback` in `testShouldHandleSOAPVersion12`? Your version is more generic and can be reused there too.